.github/workflows/integration-tests-workflow.yml

```diff
@@ -114,13 +114,29 @@ jobs:
       - name: Run OpenSearch Dashboards server
         run: |
           cd OpenSearch-Dashboards
-          yarn start --no-base-path --no-watch &
-          sleep 200
+          nohup yarn start --no-base-path --no-watch | tee dashboard.log &
+
+      - name : Check If OpenSearch Dashboards Is Ready
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          cd ./OpenSearch-Dashboards
+          echo "Start checking OpenSearch Dashboards."
+          for i in {1..60}; do
+            if grep -q "bundles compiled successfully after" "dashboard.log"; then
+              echo "OpenSearch Dashboards compiled successfully."
+              break
+            fi
+            if [ $i -eq 60 ]; then
+              echo "Timeout for 600 seconds reached. OpenSearch Dashboards did not finish compiling."
+              exit 1
+            fi
+            sleep 10
+          done
 
       - name: Install Cypress
         run: |
           cd ./OpenSearch-Dashboards/plugins/dashboards-observability
-          npx cypress install
+          npm install cypress --save-dev
```

Comment:
cypress is already defined in package.json, why is the installation a separate step with npm?

Reply:
Good catch. I was using this one only to get the latest version of cypress in my local. I have reverted this line back.